src/evaluators/state/PHAL_LoadStateField_Def.hpp

```diff
@@ -51,15 +50,13 @@ void LoadStateFieldBase<EvalT, Traits, ScalarType>::evaluateFields(typename Trai
   //       whomever changed the data.
   const auto& stateToLoad = (*workset.stateArrayPtr)[stateName];
   auto stateData = stateToLoad.dev();
-  const int stateToLoad_size = stateToLoad.size();
 
-  MDFieldVectorRight<ScalarType> g(data);
-  dataVec = g;
+  ALBANY_ASSERT (stateData.rank() <= 3, "Current implementation supports only views with rank up to 3. If larger rank is needed modify code below");
 
   Kokkos::parallel_for(this->getName(),
-                       Kokkos::RangePolicy<ExecutionSpace>(0,data.size()),
-                       KOKKOS_CLASS_LAMBDA(const int i) {
-    dataVec[i] = (i < stateToLoad_size) ? stateData(i) : 0.0;
+                       Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<3>>({0,0,0},{stateData.extent(0),stateData.extent(1),stateData.extent(2)}),
```

Comment:
This works cause we use DynRankView, so usince 3 indices is always ok (we just have extent(1) and extent(2) that may be 1), right?

Reply:
kind of weird that this works for rank<3 but nice that it does since it seems a lot simpler.